### PR TITLE
[#2651] feat(docker): optimize start.sh in Doris container

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

- Optimize`startup.sh`, cat log to stdout
- Add disk space check before starting Doris
- Exit when FE or BE start failed

### Why are the changes needed?

Fix: #2651 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

UT

### CODE REVIEW

**Comment by zhoukangcn** [comment]
@yuqi1129 could you help to review this PR when you are free.

I had build Doris container  `zhoukangcn/gravitino-ci-doris:0.1.1`, and UT in #2597 seem worked

**Comment by yuqi1129** [comment]
@zhoukangcn 
So, what's the root cause that prevents Dennis from starting successfully?

File: dev/docker/doris/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN apt-get update -y &&  DEBIAN_FRONTEND=noninteractive apt-get install -y --no
     curl wget less vim htop iproute2 numactl jq iotop sysstat tzdata xz-utils \
     tcpdump iputils-ping dnsutils strace lsof blktrace \
     bpfcc-tools linux-headers-realtime linux-tools-realtime silversearcher-ag \
-    net-tools openjdk-8-jdk && \
+    net-tools netcat openjdk-8-jdk && \
```
Comment (yuqi1129) [inline dev/docker/doris/Dockerfile:20]: Please add the change log to `docker-image-details.md`.

**Comment by zhoukangcn** [comment]
> @zhoukangcn So, what's the root cause that prevents Dennis from starting successfully?

Root Cause :
- Doris FE requires at least 6G disk space
- github workflow instance only have 5.1G  after pull doris images

so, i submit another PR #2692 to fix this problem

**Comment by yuqi1129** [comment]
It's okay with me for the change, @qqqttt123 please also help to take a look.

**Review by qqqttt123 — APPROVED** [review]
LGTM.

**Comment by qqqttt123** [comment]
New Doris ci image is released. https://github.com/datastrato/gravitino/actions/runs/8448998039 
Thanks for your great work! @zhoukangcn @yuqi1129

**Pull request merged**